src/components/Controls.vue

```diff
@@ -55,6 +59,8 @@ export default {
     this.provenance = provenance;
     this.workspace = workspace;
     this.graph = graph;
+    this.variableList = Object.keys(this.graphStructure.nodes[0]);
+    this.colorVariableList = Object.keys(this.graphStructure.nodes[0]).concat(["table"]);
```

Comment:
These should probably be computed variables.

Reply:
Doesn't it only make sense to have these be computed if they update? We don't expect that the graphStructure will change though, right?

Reply:
Computed is good for any value that is a function of other props or data props. In this case, `graphStructure` is a prop, so using a computed prop for `variableList` and `colorVariableList` is the right thing to do.

You're right that we don't expect `graphStructure` to change now, but if in the future we redesign things so that it *does* change (for instance, if we were to add a dropdown allowing us to switch which graph is being visualized, similar to how evo-bio does it), then this code will become incorrect.

One good thing about computed properties is that Vue "caches" them for you transparently. So there shouldn't be any performance impact to using them here.

Reply:
In fact, due to the argument I just made, I'd say we should definitely transfer these to computed properties before merging.

Reply:
Fixed in 326673f